### PR TITLE
iOS UITextInput autocorrection prompt

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Framework PR: https://github.com/flutter/flutter/pull/45354

Notifies the framework when `- (CGRect)firstRectForRange:(UITextRange*)range` gets called. 

Moved `FlutterTextInputView` to `FlutterTextInputPlugin.h` so it can be referenced in my xctest.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.m [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (justinmc) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.m]: Should update the year to 2019.

Comment (cbracken) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.m]: 2013 is the correct/approved year to use for all copyright dates in the engine. This has the effect of minimising our LICENSES text.

Comment (justinmc) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.m]: Whoops I have messed this up all over the place 😊 Thanks for the correction!  I'll move these back to 2013 when I see them.

**Comment by LongCatIsLooong** [comment]
Hi @cbracken I've written a migration guide and made required changes to the internal code affected. It's been a while since the PR was opened, is it still OK to merge this one?

**Comment by cbracken** [comment]
Yep -- please rebase to head and re-run presubmits if you haven't already. Once those pass you're good to go.

**Pull request merged**